src/components/AssetsPage/index.jsx

```diff
@@ -44,63 +37,41 @@ export class AssetsPage extends React.Component {
     }
   }
 
-  hasNoAssets = assetsList => (
-    assetsList.length === 0
-  )
+  getPageType = (props) => {
+    if (props.assetsList.length > 0) {
+      return pageTypes.types.NORMAL;
+    } else if (Object.keys(props.request.assetTypes).length === 0) {
+      return pageTypes.types.NO_CONTENT;
+    }
+    return pageTypes.types.NO_RESULTS;
+  }
 
-  hasAssets = assetsList => (
-    assetsList.length > 0
-  )
+  getPage = (type) => {
+    switch (type) {
+      case pageTypes.types.NORMAL:
+        return pageTypes.renderAssetsPage(this.props.uploadSettings.max_file_size_in_mbs);
```

Comment:
I'm a little iffy on this part. Originally, I just passed all of props, but then Farhanah and I settled on just passing in the necessary parameters. Just flagging this as a potential "code smell". What happens if these pages get more complex and require more data from the AssetsPage?

Reply:
AssetsPage isn't much of a component without the pageTypes. I wonder if it would make things simpler to just have those render*Page functions defined on the AssetsPage class so they have access to whatever props that they need via `this.props`.

Reply:
@thallada That's actually what I had originally, but there was some concern about the number of functions in AssetsPage (all of pageTypes.jsx was in index.jsx). I'm happy to move it back, as it would live closer to the data it needs, but I'm not sure how to prevent the file from being enormous. Do you have a preference? I'd actually prefer to move it back, but if there are good arguments for separating it out, I'm happy to keep it.

Reply:
I somewhat prefer to have the functions in this class, but I can see the argument for separating them too :man_shrugging: . I think it doesn't matter too much either way.